### PR TITLE
[sil.rst] Try to make it clearer that copy_unmanaged_value performs a…

### PR DESCRIPTION
**Pull request by gottesmm**
… "strong copy" of the underlying value.

### CODE REVIEW

**Comment by gottesmm** [comment]
@swift-ci smoke test

**Comment by gottesmm** [comment]
@atrick is this clearer?

**Comment by jrose-apple** [comment]
Too bad `copy_from_unmanaged_value` doesn't really fit the naming conventions.

**Review by atrick — APPROVED** [review]
lgtm

**Comment by atrick** [comment]
Yeah, @jrose-apple hit upon why I was confused by the instruction name. Copying an `unmanaged` value would not produce a retain. In fact, if the comments just said "this copies an unmanaged value into a managed value, retaining the managed value" that would be clear to me.

**Comment by gottesmm** [comment]
@atrick @jrose-apple I was thinking of renaming this to strong_copy_unmanaged_value. Would that be a better name?

**Comment by gottesmm** [comment]
(I would also rename copy_unowned_value to strong_copy_unowned_value if that makes sense). The reason strong may make sense here is due to the precedent of strong_retain_unowned.

**Comment by gottesmm** [comment]
(which is take a "safe unowned" value, check that it has not been destroyed and then bump the ref count. It is semantically a conversion from "safe unowned" -> strong).

**Comment by jrose-apple** [comment]
Sure, that works for me.

**Comment by atrick** [comment]
@gottesmm can this be applied to an `@weak` type? That wouldn't really make sense then.

**Comment by gottesmm** [comment]
No. @weak types are modeled as a storage location that you load from. See:

https://github.com/apple/swift/blob/master/docs/SIL.rst#load-weak

We do not represent them as SSA values since they are address only types.

**Comment by gottesmm** [comment]
Sorry, just saw this was an old PR. Let me fix this real quick.

**Comment by atrick** [comment]
@gottesmm let me rephrase: "strong" should only be used in the opcode name when we have a concrete type that's known to be a strong reference (or contain strong references). If copy-from-unmanaged asserts that, then the name is fine.
